### DISCUSSION TITLE
inject sidecar

### DISCUSSION THREAD

**Discussion by Jerry-yz** [istio/istio#52337] [Q&A]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

when i use istio inject sidecar, i have ns label istio-injection:enabled
kubectl get ns microservice -o jsonpath='{.metadata.labels}'
{"istio-injection":"enabled","kubernetes.io/metadata.name":"microservice"}
i create a deployment and labels pod  sidecar.istio.io/inject: "false", pod is created, it has sidecar container
this is current？

### Version

```Text
istioctl version
client version: 1.17.2
control plane version: 1.17.2
data plane version: 1.17.2 (7 proxies)
kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.17", GitCommit:"953be8927218ec8067e1af2641e540238ffd7576", GitTreeState:"clean", BuildDate:"2023-02-22T13:34:27Z", GoVersion:"go1.19.6", Compiler:"gc", Platform:"linux/arm64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.17", GitCommit:"953be8927218ec8067e1af2641e540238ffd7576", GitTreeState:"clean", BuildDate:"2023-02-22T13:27:46Z", GoVersion:"go1.19.6", Compiler:"gc", Platform:"linux/arm64"}
```

**Comment by howardjohn** [comment]
Are you sre you are labelling the pod, not the deployment? deployment.spec.template.metadata, not deployment.metadata

**Comment by Jerry-yz** [comment]
> Are you sre you are labelling the pod, not the deployment? deployment.spec.template.metadata, not deployment.metadata

yes, i am sure
kubectl get pod -n microservice store-867b9cf54c-qtmhg -o yaml
![Uploading image.png…]()

**Comment by howardjohn** [comment]
Looks like the image did not upload
